stdlib/public/core/Array.swift

```diff
@@ -1311,6 +1311,12 @@ extension Array: RangeReplaceableCollection {
     }
     return _copyCollectionToContiguousArray(self)
   }
+  
+  @inlinable
+  public mutating func swapAt(_ i: Int, _ j: Int) {
+    _makeUniqueAndReserveCapacityIfNotUnique()
```

Comment:
What is the cost of this call if `i == j`? You check for that in `ContiguousArrayBuffer.swapAt(_:_:)`, but at that point has the copy already been made?